File: JAVA/Figures exercise/com/figures/FigureGeometric.java
package com.figures;


public abstract class FigureGeometric {
    private final String name;
    private final String nameSpanish;

    public FigureGeometric(String name, String nameSpanish) {
        this.name = name;
        this.nameSpanish = nameSpanish;
    }
    
    public String getName(){
        return this.name;
    }
    public String getNameSpanish(){
        return this.nameSpanish;
    }

    public abstract double calculateArea();
    
}

File: PHP/Figures exercise/figures/FigureGeometric.php
<?php
abstract class FigureGeometric{
    private String $name;
   
    
    public function _construct ( String $name){
        $this->name= $name;

    }
    public function getName():String{
        return $this->name;
    }

    
    public abstract function calculateArea():float;
}